internal/pkg/manifest/validate.go

```diff
@@ -895,6 +902,17 @@ func (r RoutingRule) validate() error {
 			conditionalFields: []string{"hosted_zone"},
 		}
 	}
+	aliases, err := r.Alias.ToStringSlice()
+	if err != nil {
+		return fmt.Errorf("convert aliases to string slice: %w", err)
+	}
+	allowedSourceIps := make([]string, len(r.AllowedSourceIps))
+	for idx, ip := range r.AllowedSourceIps {
+		allowedSourceIps[idx] = string(ip)
+	}
+	if err := validateConditionValuesPerRule(aliases, allowedSourceIps); err != nil {
```

Comment:
I kept this `typecast to string` as is, because I want to use `english.WordSeries` and also considering this [logic](https://github.com/aws/copilot-cli/pull/4961/files#diff-5fc13795122fe030799a09ef956a8b66cec3d40bf1a81266323edbf3626b5685R2343-R2349) where I need to have both aliases and allowedsourceIps to be `[]string`